### PR TITLE
✨ Inactivity (front-end part)

### PR DESCRIPTION
**Pull request by ignapas**
## What do these changes do?
* Makes use of user preference `userInactivityThreshold`
* If there is no activity (Qooxdoo front-end keyboard and mouse) during that time, it will start asking the `inactivity` endpoint, every minute, whether it is possible to close the study.

## Related issue/s
ITISFoundation/osparc-issues#953
ITISFoundation/osparc-issues#1153

## How to test
Tricky. To my knowledge, only S4L will implement this for the moment. It also depends on [this](https://github.com/ITISFoundation/osparc-simcore/pull/5127) to be merged first. I would wait until it is useful and testable to merge it.

## Dev Checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: services/static-webserver/client/source/class/osparc/desktop/StudyEditorIdlingTracker.js [hunk 1]
```diff
@@ -63,16 +64,40 @@ qx.Class.define("osparc.desktop.StudyEditorIdlingTracker", {
...
+          if (!this.__frontendTimedout) {
+            // User was inactive in oSPARC, we can test services (ask every minute until response is true or we detect activity again)
+            this.__removeIdleFlashMessage();
+            const checkInactivity = () => {
+              osparc.data.Resources.fetch("studies", "getInactivity", {
+                url: {
+                  studyId: this.__studyUuid
+                }
+              }).then(({data}) => {
+                if (data) {
```
Comment (GitHK) [inline services/static-webserver/client/source/class/osparc/desktop/StudyEditorIdlingTracker.js:81]: Actually inside the data, it provides an "is_inactive" flag that you are supposed to check as well

Comment (GitHK) [reply services/static-webserver/client/source/class/osparc/desktop/StudyEditorIdlingTracker.js:81]: As soon as `is_inactive` is `true` you can stop checking.

**Review by jsaq007 — CHANGES_REQUESTED** [review]
Linter and other review items

File: services/static-webserver/client/source/class/osparc/desktop/StudyEditorIdlingTracker.js (hunk 1 above)
Comment (ignapas) [reply services/static-webserver/client/source/class/osparc/desktop/StudyEditorIdlingTracker.js]: 👍🏼

File: services/static-webserver/client/source/class/osparc/product/Utils.js [hunk 2]
```diff
@@ -32,6 +32,15 @@ qx.Class.define("osparc.product.Utils", {
       return (productName === product);
     },
 
+    isS4LProduct: function() {
```
Comment (GitHK) [inline services/static-webserver/client/source/class/osparc/product/Utils.js:35]: 👍

Comment (GitHK) [reply services/static-webserver/client/source/class/osparc/product/Utils.js:35]: maybe call them s4lBasedProducts since, there are multiple.

**Review by GitHK — APPROVED** [review]

**Review by mguidon — APPROVED** [review]
I tested this and it works on gsparc. We might reconsider this to be availabe in other products.

**Review by jsaq007 — APPROVED** [review]
Code looks good... For the moment I need to trust it works in the browser.

**Pull request merged**